OpenSim/Examples/DataTable/example.cpp

```diff
@@ -0,0 +1,357 @@
+  SimTK::Vector_<SimTK::Real> col_real{static_cast<int>(data2_real.size()),
+                                       data2_real.data()};
+  SimTK::Vector_<SimTK::Vec3> col_vec3{static_cast<int>(data2_vec3.size()),
+                                       data2_vec3.data()};
+  SimTK::Vector_<SimTK::Vec6> col_vec6{static_cast<int>(data2_vec6.size()),
+                                       data2_vec6.data()};
+  dt_real.addCol(col_real);
+  dt_vec3.addCol(col_vec3);
+  dt_vec6.addCol(col_vec6);
+
+  // Add a column using an iterator.
+  dt_real.addCol(data2_real.cbegin(), data2_real.cend());
+  dt_vec3.addCol(data2_vec3.cbegin(), data2_vec3.cend());
+  dt_vec6.addCol(data2_vec6.cbegin(), data2_vec6.cend());
+
+  // Add multiple columns using an iterator.
+  for(int i = 0; i < 7; ++i) {
+    data2_real.push_back(i);
+    data2_vec3.push_back(SimTK::Vec3{1, 2, 3} + i);
+    data2_vec6.push_back(SimTK::Vec6{1, 2, 3, 4, 5, 6} + i);
+  }
+  dt_real.addCols(data2_real.cbegin(), data2_real.cend());
+  dt_vec3.addCols(data2_vec3.cbegin(), data2_vec3.cend());
+  dt_vec6.addCols(data2_vec6.cbegin(), data2_vec6.cend());
+
+  // Copy construct DataTable.
+  decltype(dt_real) dt_real_copy{dt_real};
+  decltype(dt_vec3) dt_vec3_copy{dt_vec3};
+  decltype(dt_vec6) dt_vec6_copy{dt_vec6};
+
+  // Add/bind an entire DataTable by row.
+  dt_real.rbindDataTable(dt_real_copy);
+  dt_vec3.rbindDataTable(dt_vec3_copy);
+  dt_vec6.rbindDataTable(dt_vec6_copy);
+
+  // Copy assign DataTable.
+  dt_real_copy = dt_real;
+  dt_vec3_copy = dt_vec3;
+  dt_vec6_copy = dt_vec6;
+
+  // Add/bind an entire DataTable by col.
+  dt_real.cbindDataTable(dt_real_copy);
+  dt_vec3.cbindDataTable(dt_vec3_copy);
+  dt_vec6.cbindDataTable(dt_vec6_copy);
+
+  // Bind two existing DataTable(s) and produce a new one instead of appending
+  // one to another.
+  {
+    // Construct DataTable of a given size with a default value.
+    decltype(dt_real) dt1_real{3, 4, SimTK::Real{10}};
+    decltype(dt_vec3) dt1_vec3{3, 4, SimTK::Vec3{10, 20, 30}};
+    decltype(dt_vec6) dt1_vec6{3, 4, SimTK::Vec6{10, 20, 30, 40, 50, 60}};
+
+    // Construct DataTable of a given size with a default value.
+    decltype(dt_real) dt2_real{6, 4, SimTK::Real{5}};
+    decltype(dt_vec3) dt2_vec3{6, 4, SimTK::Vec3{5, 6, 7}};
+    decltype(dt_vec6) dt2_vec6{6, 4, SimTK::Vec6{5, 6, 7, 8, 9, 0}};
+
+    // Bind by row.
+    auto rbind_dt_real = rbindDataTables(dt1_real, dt2_real);
+    auto rbind_dt_vec3 = rbindDataTables(dt1_vec3, dt2_vec3);
+    auto rbind_dt_vec6 = rbindDataTables(dt1_vec6, dt2_vec6);
+
+    // Bind by col. The following using same table for both arguments.
+    auto cbind_dt_real = cbindDataTables(dt1_real, dt1_real);
+    auto cbind_dt_vec3 = cbindDataTables(dt1_vec3, dt1_vec3);
+    auto cbind_dt_vec6 = cbindDataTables(dt1_vec6, dt1_vec6);
+  }
+
+  // The columns of a DataTable can be labeled. Not all columns need to be
+  // labeled. 
+  dt_real.insertColLabel(0, "col-zero");
```

Comment:
`insert` implies that something gets moved out of the way. This looks more like a `set` operation to me.